src/components/learner-credit-management/AssignmentStatusTableCell.jsx

```diff
@@ -9,13 +11,49 @@ import NotifyingLearner from './assignments-status-chips/NotifyingLearner';
 import WaitingForLearner from './assignments-status-chips/WaitingForLearner';
 import { capitalizeFirstLetter } from '../../utils';
 
-const AssignmentStatusTableCell = ({ row }) => {
+const AssignmentStatusTableCell = ({ enterpriseId, row }) => {
   const { original } = row;
   const {
     learnerEmail,
     learnerState,
     errorReason,
   } = original;
+  const sharedTrackEventMetadata = {
+    learnerState,
+  };
+
+  const sendGenericTrackEvent = (eventName, eventMetadata = {}) => {
+    sendEnterpriseTrackEvent(
+      enterpriseId,
+      eventName,
+      {
+        ...sharedTrackEventMetadata,
+        ...eventMetadata,
+      },
+    );
+  };
+
+  const sendErrorStateTrackEvent = (eventName, eventMetadata = {}) => {
+    const errorReasonMetadata = !errorReason
+      ? {
+        errorReason: null,
+      }
+      : {
+        errorReason: errorReason.errorReason,
+        actionType: errorReason.actionType,
+      };
+    const errorStateMetadata = {
+      ...sharedTrackEventMetadata,
+      ...errorReasonMetadata,
+      ...eventMetadata,
+    };
```

Comment:
I think this would be pretty confusing on the receiving end because the resulting payload would contain an `actionType` key, but only when there's an error, and it's the type for the action that failed.  This is not self-evident.  The key name should be something like "failedActionType" instead.

Also, i'm not sure how the eventName is being passed to this function since it's called without parameters below.

Reply:
Ended up with a uniqued object with your suggestion, ended up with a cleaner result 👍🏽